src/engine/SmtCore.h

```diff
@@ -98,6 +100,16 @@ class SmtCore
 
     void setConstraintViolationThreshold( unsigned threshold );
 
+    /*
+      Replay a stackEntry
+    */
+    void replayStackEntry( StackEntry *stackEntry );
+
+    /*
+      Store the current stack into smtState
+    */
+    void storeSmtState( SmtState &smtState );
```

Comment:
The lack of symmetry here is unclear. Why not have "store" and "restore" methods? Why do you want to replay each stack entry individually? That would be useful only if you wanted to replay some but not others - is this the case? Right now, the logic is at the Engine instead of the SmtCore.